### ISSUE TITLE
pixman_region32_init_rect: Invalid rectangle passed

### ISSUE THREAD

**Issue by camilstaps** [fpaquet/gtksheet#15]
I can get GtkSheet working fine, but keep getting bug warnings on `stderr`:

```
*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug
```

This is from the example on https://fpaquet.github.io/gtksheet/tutorial/x39.html, which I adapted to Gtk 3 as follows:

```
#include <gtk/gtk.h>
#include <gtksheet/gtksheet.h>

int main(int argc, char **argv) {
	gtk_init(&argc, &argv);

	GtkWidget *window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
	gtk_window_set_title(GTK_WINDOW(window), "GTK Sheet");
	gtk_widget_set_size_request(GTK_WIDGET(window), 900, 600);

	GtkWidget *box = gtk_box_new(GTK_ORIENTATION_VERTICAL, 1);
	gtk_box_set_homogeneous(GTK_BOX(box), TRUE);
	gtk_container_add(GTK_CONTAINER(window), box);

	GtkWidget *scrolled_window = gtk_scrolled_window_new(NULL, NULL);
	gtk_box_pack_start(GTK_BOX(box), scrolled_window, TRUE, TRUE, 1);

	GtkWidget *sheet = gtk_sheet_new(1000, 26, "Sheet 1");
	gtk_sheet_set_autoresize(GTK_SHEET(sheet), TRUE);
	gtk_container_add(GTK_CONTAINER(scrolled_window), sheet);

	gtk_widget_show_all(window);

	g_signal_connect(window, "destroy", G_CALLBACK (gtk_main_quit), NULL);

	gtk_main();

	return 0;
}
```

After installing `libpixman-1-0-dbgsym` (with `deb http://deb.debian.org/debian-debug/ buster-debug main` in my `/etc/apt/sources.list`; I'm on Debian Buster) I can get a backtrace with `gdb`, but this doesn't tell me very much:

```
$ gdb ./test -ex 'b _pixman_log_error' -ex r -ex bt
GNU gdb (Debian 8.2.1-2+b1) 8.2.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./test...done.
Function "_pixman_log_error" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (_pixman_log_error) pending.
Starting program: /home/camil/projects/text-sheets/src/test 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff4a95700 (LWP 2241)]
[New Thread 0x7fffeffff700 (LWP 2242)]

Thread 1 "test" hit Breakpoint 1, _pixman_log_error (
    function=function@entry=0x7ffff5f38c90 <__PRETTY_FUNCTION__.5039> "pixman_region32_init_rect", 
    message=message@entry=0x7ffff5f38977 "Invalid rectangle passed") at ../../pixman/pixman-utils.c:320
320	../../pixman/pixman-utils.c: No such file or directory.
#0  0x00007ffff5f08af0 in _pixman_log_error
    (function=function@entry=0x7ffff5f38c90 <__PRETTY_FUNCTION__.5039> "pixman_region32_init_rect", message=message@entry=0x7ffff5f38977 "Invalid rectangle passed") at ../../pixman/pixman-utils.c:320
#1  0x00007ffff5f05d9a in pixman_region32_init_rect
    (region=0x555555796d58, x=<optimized out>, y=<optimized out>, width=<optimized out>, height=<optimized out>)
    at ../../pixman/pixman-region.c:389
#2  0x00007ffff7620388 in cairo_region_create_rectangle () at /usr/lib/x86_64-linux-gnu/libcairo.so.2
#3  0x00007ffff77bb550 in recompute_visible_regions_internal
    (private=0x55555589d010, recalculate_clip=recalculate_clip@entry=1, recalculate_children=recalculate_children@entry=0)
    at ../../../../gdk/gdkwindow.c:1018
#4  0x00007ffff77bb7cc in recompute_visible_regions (private=<optimized out>, recalculate_children=recalculate_children@entry=0)
    at ../../../../gdk/gdkwindow.c:1119
#5  0x00007ffff77bbea7 in set_viewable (w=0x55555589d010, val=val@entry=1) at ../../../../gdk/gdkwindow.c:5202
#6  0x00007ffff77bbe4a in set_viewable (w=0x555555592c80, val=val@entry=1) at ../../../../gdk/gdkwindow.c:5210
#7  0x00007ffff77bbe4a in set_viewable (w=0x5555555924b0, val=val@entry=1) at ../../../../gdk/gdkwindow.c:5210
#8  0x00007ffff77bbe4a in set_viewable (w=0x555555592320, val=1) at ../../../../gdk/gdkwindow.c:5210
#9  0x00007ffff77bbefa in _gdk_window_update_viewable (window=window@entry=0x555555592320) at ../../../../gdk/gdkwindow.c:5267
#10 0x00007ffff77a90c8 in _gdk_set_window_state (window=0x555555592320, new_state=GDK_WINDOW_STATE_FOCUSED)
    at ../../../../gdk/gdkevents.c:2252
#11 0x00007ffff77c0fda in gdk_window_show_internal (raise=<optimized out>, window=0x555555592320) at ../../../../gdk/gdkwindow.c:5294
#12 0x00007ffff77c0fda in gdk_window_show_internal (window=0x555555592320, raise=<optimized out>) at ../../../../gdk/gdkwindow.c:5271
#13 0x00007ffff7c07d49 in gtk_window_map (widget=0x5555555ec290) at ../../../../gtk/gtkwindow.c:6328
#14 0x00007ffff7393ec6 in  () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#15 0x00007ffff73b038d in g_signal_emit_valist () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#16 0x00007ffff73b097f in g_signal_emit () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#17 0x00007ffff7bf2002 in gtk_widget_map (widget=0x5555555ec290) at ../../../../gtk/gtkwidget.c:4999
#18 0x00007ffff7bff8ad in gtk_window_show (widget=0x5555555ec290) at ../../../../gtk/gtkwindow.c:6184
#19 0x00007ffff7393c8d in g_closure_invoke () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#20 0x00007ffff73a74b4 in  () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#21 0x00007ffff73b02be in g_signal_emit_valist () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#22 0x00007ffff73b097f in g_signal_emit () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#23 0x00007ffff7bebee6 in gtk_widget_show (widget=0x5555555ec290) at ../../../../gtk/gtkwidget.c:4800
#24 0x00007ffff7bebee6 in gtk_widget_show (widget=0x5555555ec290) at ../../../../gtk/gtkwidget.c:4773
#25 0x0000555555555400 in main (argc=1, argv=0x7fffffffddd8) at test.c:22
```

Any ideas?

System details:

- amd64 Debian Buster
- X11 (`xorg 1:7.7+19`)
- i3 window manager (`i3-wm 4.16.1-1`)
- Gtk `3.24.5-1`

**Comment by benreu** [comment]
I also have this problem

Linux Mint 64 bit
Cinnamon window manager
Gtk 3.22

**Comment by wmclaugh** [comment]
The issue appears to occur in release version 4.3.3 due to negative values being passed in for the height/width of rectangles in _gtk_sheet_row_buttons_size_allocate and  _gtk_sheet_column_buttons_size_allocate. See https://github.com/fpaquet/gtksheet/pull/21 for code changes that should prevent the warnings.

**Comment by fpaquet** [comment]
Somewhat late, but i didn't get notifications for the sheet issues.

Looks like a bug, i will check proposed fix during next development session.

**Comment by fpaquet** [comment]
Merged pull request #21 from wmclaugh/issue15_workaround
